account.go

```diff
@@ -0,0 +1,160 @@
+package caigo
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+)
+
+type Signer struct {
+	private *big.Int
+	Curve   StarkCurve
+	Gateway *StarknetGateway
+	PublicX *big.Int
+	PublicY *big.Int
+}
+
+type FeeEstimate struct {
+	Amount *big.Int `json:"amount"`
+	Unit   string   `json:"unit"`
+}
+
+/*
+	Instantiate a new StarkNet Signer which includes structures for calling the network and signing transactions:
+	- private signing key
+	- stark curve definition
+	- full StarknetGateway definition
+	- public key pair for signature verifications
+*/
+func (sc StarkCurve) NewSigner(private, pubX, pubY *big.Int, chainId ...string) (signer *Signer, err error) {
+	if len(sc.ConstantPoints) == 0 {
+		return signer, fmt.Errorf("must initiate precomputed constant points")
+	}
+	var gw *StarknetGateway
```

Comment:
I think it would be more intuitive to pass in the provider than instantiate it internally. There are configuration options that cant be given here and eventually provider should be an interface, in which case others implementations easily be passed in

Reply:
agreed